dipy/denoise/nlmeans_block.pyx

```diff
@@ -277,25 +278,25 @@ cdef double _local_mean(double[:, :, :]ima, int x, int y, int z) nogil:
                                           pz - 1 if pz >= dims2 else pz))
                 ss += ima[px, py, pz]
 
-    return ss / 27.0
+    return ss / vol
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 @cython.cdivision(True)
-cdef double _local_variance(double[:, :, :] ima, double mean, int x, int y, int z) nogil:
+cdef double _local_variance(double[:, :, :] ima, double mean, int x, int y, int z, int p) nogil:
     """
-    local variance of a 3x3x3 patch centered at x,y,z
+    local variance of a patch with patch radius = p centered at x,y,z
     """
     dims0 = ima.shape[0]
     dims1 = ima.shape[1]
     dims2 = ima.shape[2]
     cdef int cnt = 0
     cdef double ss = 0
     cdef int dx, dy, dz, nx, ny, nz
-    for px in range(x - 1, x + 2):
-        for py in range(y - 1, y + 2):
-            for pz in range(z - 1, z + 2):
+    for px in range(x - p, x + p + 1):
+        for py in range(y - p, y + p + 1):
+            for pz in range(z - p, z + p + 1):
                 if ((px >= 0 and py >= 0 and pz > 0) and
                         (px < dims0 and py < dims1 and pz < dims2)):
                     ss += (ima[px, py, pz] - mean) * (ima[px, py, pz] - mean)
```

Comment:
Variance is normalized by the count variable which is essentially the volume, just a gimmick of the code